articles/introduction-to-r/index.md

```diff
@@ -0,0 +1,328 @@
+
+```r
+matrix(1:16, bycol=TRUE, ncol=4)
+matrix(1:16, byrow=TRUE, nrow=4)
+```
+
+If the vector's length is not a multiple of the number of rows/columns mentioned in the function, then R displays a warning and fills up the remaining spaces with the first few elements of the list.
+
+For example
+
+```r
+matrix(1:16, byrow=TRUE, nrow=3 )
+```
+The above code generates the following:
+```txt
+Warning message: data length [16] is not a sub-multiple or multiple of the number of rows [3]
+     [,1] [,2] [,3] [,4] [,5] [,6]
+[1,]    1    2    3    4    5    6
+[2,]    7    8    9   10   11   12
+[3,]   13   14   15   16    1    2
+```
+##### Naming Matrices
+
+Just like vectors, matrices also have the option to name the rows and columns. The functions available are `rownames()` and `colnames()` respectively. They take in a vector containing the names corresponding to the indices. Consider the following example:
+
+```r
+new_matrix <- matrix(1:4, byrow=TRUE, ncol=2)
+colnames(newm_matrix) <- c("column 1","column 2","column 3")
+rownames(newm_matrix) <- c("row 1","row 2","row 3")
+```
+##### Summing Entries
+
+Let's say you want to compute the average of the entries in your matrix. We begin by adding them up and then dividing by the total number of entries. R offers functions to add the entire rows or columns. They are `rowSums,` `colSums.` These functions take the matrix as their input and output a vector containing the sums.
+
+Example:
+```r
+x <- matrix(1:9, byrow = TRUE, nrow=3)
+row_sum <- rowSums(x)
+col_sum <- colSums(x)
+```
+
+##### Inserting new rows and columns to existing matrices
+
+Let us consider the situation where we need to modify a matrix by adding extra rows or columns. R provides two functions called `cbind()` and `rbind()` to do the same.
+The input to the function is the matrix to be modified followed by the vector to be inserted. Consider the example given below:
+```r
+x <- matrix(1:9, byrow = TRUE, nrow=3)
+new_matrix_with_additional_column <- cbind(x, c(1,2,3))
+new_matrix_with_additional_row <- rbind(new_matrix_with_additional_column, c(4,5,6))
+new_matrix_with_additional_row
+```
+The ouput for the above code is given as follows:
+```txt
+ new_matrix_with_additional_row
+     [,1] [,2] [,3] [,4]
+[1,]    1    2    3    1
+[2,]    4    5    6    2
+[3,]    7    8    9    3
+[4,]    4    5    6    4
+```
+
+##### Selecting Elements
+The final concept to be covered under matrices is the selection of elements. Matrices in R work like 2-D arrays in other programming languages. Therefore all the slicing operations stand true in R. Let us assume we have a `4X4` dimensional matrix. The first quarter is a `2X2` matrix. To get the first quarter of the matrix, we will use the following code:
+
+```r
+x <- matrix(1:16, brow=TRUE, nrow=4) # matrix definition
```

Comment:
You need to explain brow and nrow, and again- what dimensions is this supposed to yield

Reply:
1:16 is not explained anywhere either - of how this takes all digits from 1 to 16 and populates the matrix